jormungandr/src/fragment/logs.rs

```diff
@@ -57,15 +57,21 @@ impl Logs {
                 }
             }
             None => {
-                // while a log modification, if the log was not already present in the
-                // logs it means we received it from the a new block from the network.
-                // we can mark the status of the transaction so newly received transaction
-                // be stored.
-
-                self.entries.put(
-                    fragment_id,
-                    FragmentLog::new(fragment_id.clone().into_hash(), FragmentOrigin::Network),
-                );
+                // Possible reasons for entering this branch are:
+                //
+                // - Receiving a fragment from the network.
```

Comment:
I think a fragment received from the network would not be fed into this function at its first step, but rather inserted as pending before normally.
What could happen instead is that we try to remove all fragments in a block received from the network, but we never received some of the fragments inside it.